controllers/activemqartemis_controller_cert_manager_test.go

```diff
@@ -0,0 +1,292 @@
+}
+
+var _ = Describe("artemis controller with cert manager test", Label("controller-cert-mgr-test"), func() {
+	var installedCertManager bool = false
+
+	BeforeEach(func() {
+		if os.Getenv("USE_EXISTING_CLUSTER") == "true" {
+			//if cert manager/trust manager is not installed, install it
+			if !CertManagerInstalled() {
+				Expect(InstallCertManager()).To(Succeed())
+				installedCertManager = true
+			}
+
+			rootIssuer = InstallClusteredIssuer(rootIssuerName, nil)
+
+			rootCert = InstallCert(rootCertName, rootCertNamespce, func(candidate *cmv1.Certificate) {
+				candidate.Spec.IsCA = true
+				candidate.Spec.CommonName = "artemis.root.ca"
+				candidate.Spec.SecretName = rootCertSecretName
+				candidate.Spec.IssuerRef = cmmetav1.ObjectReference{
+					Name: rootIssuer.Name,
+					Kind: "ClusterIssuer",
+				}
+			})
+
+			caIssuer = InstallClusteredIssuer(caIssuerName, func(candidate *cmv1.ClusterIssuer) {
+				candidate.Spec.SelfSigned = nil
+				candidate.Spec.CA = &cmv1.CAIssuer{
+					SecretName: rootCertSecretName,
+				}
+			})
+			InstallCaBundle(caBundleName, rootCertSecretName, caPemTrustStoreName)
+		}
+	})
+
+	AfterEach(func() {
+		if os.Getenv("USE_EXISTING_CLUSTER") == "true" {
+			UnInstallCaBundle(caBundleName)
+			UninstallClusteredIssuer(caIssuerName)
+			UninstallCert(rootCert.Name, rootCert.Namespace)
+			UninstallClusteredIssuer(rootIssuerName)
+
+			if installedCertManager {
+				Expect(UninstallCertManager()).To(Succeed())
+				installedCertManager = false
+			}
+		}
+	})
+
+	Describe("cert manager integration test", func() {
+		Context("tls exposure with cert manager", func() {
+			BeforeEach(func() {
+				if os.Getenv("USE_EXISTING_CLUSTER") == "true" {
+					InstallCert(serverCert, defaultNamespace, func(candidate *cmv1.Certificate) {
+						candidate.Spec.DNSNames = []string{brokerCrNameBase + "0-ss-0", brokerCrNameBase + "1-ss-0", brokerCrNameBase + "2-ss-0"}
+						candidate.Spec.IssuerRef = cmmetav1.ObjectReference{
+							Name: caIssuer.Name,
+							Kind: "ClusterIssuer",
+						}
+					})
+				}
+			})
+			AfterEach(func() {
+				if os.Getenv("USE_EXISTING_CLUSTER") == "true" {
+					UninstallCert(serverCert, defaultNamespace)
+				}
+			})
+			It("test configured with cert and ca bundle", func() {
+				if os.Getenv("USE_EXISTING_CLUSTER") == "true" {
+					testConfiguredWithCertAndBundle(serverCert+"-secret", caBundleName)
+				}
+			})
+		})
+	})
+})
+
+func getConnectorConfig(podName string, crName string, connectorName string, g Gomega) map[string]string {
+	curlUrl := "http://" + podName + ":8161/console/jolokia/read/org.apache.activemq.artemis:broker=\"amq-broker\"/ConnectorsAsJSON"
+	command := []string{"curl", "-k", "-s", "-u", "testuser:testpassword", curlUrl}
+
+	result := ExecOnPod(podName, crName, defaultNamespace, command, g)
+
+	var rootMap map[string]any
+	g.Expect(json.Unmarshal([]byte(result), &rootMap)).To(Succeed())
+	connectors := rootMap["value"].(string)
```

Comment:
The test should check if `rootMap["value"]` is nil to avoid panic errors, i.e.
```suggestion
       rootMapValue := rootMap["value"]
       g.Expect(rootMapValue).ShouldNot(BeNil())
       connectors := rootMapValue.(string)
```

Reply:
+1